do/apps.go

```diff
@@ -119,6 +121,86 @@ func (s *appsService) Delete(appID string) error {
 	return err
 }
 
+func (s *appsService) Detect(source string, sha string, name string, branch string, autoDeploy bool) (*godo.AppSpec, error) {
+	var dr godo.DetectRequest
+	if strings.Contains(source, "github") {
+		dr.GitHub = &godo.GitHubSourceSpec{
+			Repo:         verifyGitSource(source, "github"),
+			Branch:       branch,
+			DeployOnPush: autoDeploy,
+		}
+	} else if strings.Contains(source, "gitlab") {
+		dr.GitLab = &godo.GitLabSourceSpec{
+			Repo:         verifyGitSource(source, "gitlab"),
+			Branch:       branch,
+			DeployOnPush: autoDeploy,
+		}
+	} else {
+		dr.Git = &godo.GitSourceSpec{
+			RepoCloneURL: source,
+			Branch:       branch,
+		}
+	}
+	dr.SourceDir = "/"
+	dr.CommitSHA = sha
+
+	resp, _, err := s.client.Apps.Detect(context.Background(), &dr)
+	if err != nil {
+		return nil, err
+	}
+
+	var appSpec godo.AppSpec
+
+	appSpec.Name = name
+	var funcSpecArray []*godo.AppFunctionsSpec
+	for _, component := range resp.Components {
+
+		if component.Strategy == "SERVERLESS" {
```

Comment:
We should print out a warning if we detect non serverless component strategies for now until we support them.